tsfile/src/main/java/org/apache/iotdb/tsfile/read/common/TimeColumn.java

```diff
@@ -19,60 +19,115 @@
 package org.apache.iotdb.tsfile.read.common;
 
 
+import java.nio.ReadOnlyBufferException;
+import java.util.ArrayList;
+import org.apache.iotdb.tsfile.common.conf.TSFileDescriptor;
+
 public class TimeColumn {
 
-  private static final int DEFAULT_INIT_SIZE = 1000;
+  private static final int capacityThreshold = TSFileDescriptor.getInstance().getConfig()
+      .getBatchSize();
+  private int capacity = 16;
 
+  // outer list index for read
+  private int readCurListIndex;
+  // inner array index for read
+  private int readCurArrayIndex;
 
-  private long[] times;
+  // outer list index for write
+  private int writeCurListIndex;
+  // inner array index for write
+  private int writeCurArrayIndex;
 
-  private int size;
+  // the insert timestamp number of timeRet
+  private int count;
 
-  private int cur;
+  private ArrayList<long[]> timeRet;
 
   public TimeColumn() {
-    this(DEFAULT_INIT_SIZE);
+    this.readCurListIndex = 0;
+    this.readCurArrayIndex = 0;
+    this.writeCurListIndex = 0;
+    this.writeCurArrayIndex = 0;
+    timeRet = new ArrayList<>();
+    timeRet.add(new long[capacity]);
+    count = 0;
   }
 
-  public TimeColumn(int initSize) {
-    times = new long[initSize];
-  }
+  public TimeColumn(ArrayList<long[]> timeRet, int count) {
+    this.count = count;
+    this.readCurListIndex = 0;
+    this.readCurArrayIndex = 0;
 
+    while (capacity < capacityThreshold) {
+      capacity <<= 1;
+    }
 
-  public TimeColumn(long[] times) {
-    this.times = times;
+    this.writeCurListIndex = count / capacity;
+    this.writeCurArrayIndex = count & (capacity - 1);
+    this.timeRet = timeRet;
   }
 
   public void add(long time) {
-    if (size == times.length) {
-      long[] newArray = new long[times.length * 2];
-      System.arraycopy(times, 0, newArray, 0, times.length);
-      times = newArray;
+    if (writeCurArrayIndex == capacity) {
+      if (capacity >= capacityThreshold) {
+        timeRet.add(new long[capacity]);
+        writeCurListIndex++;
+        writeCurArrayIndex = 0;
+      } else {
+        int newCapacity = capacity << 1;
+
+        long[] newTimeData = new long[newCapacity];
+        System.arraycopy(timeRet.get(0), 0, newTimeData, 0, capacity);
+        timeRet.set(0, newTimeData);
+
+        capacity = newCapacity;
+      }
     }
-    times[size++] = time;
-  }
-
-  public long[] getTimes() {
-    return times;
+    timeRet.get(writeCurListIndex)[writeCurArrayIndex] = time;
+    writeCurArrayIndex++;
+    count++;
   }
 
   public boolean hasCurrent() {
-    return size > 0 && cur < size;
+    if (readCurListIndex < writeCurListIndex) {
+      return readCurArrayIndex < capacity;
+    } else if (readCurListIndex == writeCurListIndex) {
+      return readCurArrayIndex < writeCurArrayIndex;
+    } else {
+      return false;
+    }
```

Comment:
could be simpliy:
```suggestion
    } else {
      return readCurListIndex == writeCurListIndex && readCurArrayIndex < writeCurArrayIndex;
    }
```

Reply:
excellent advice,  exchange order will be more easy to understand